ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/contentassist/BuildPropertiesContentAssistProcessor.java

```diff
@@ -44,13 +48,35 @@ public ICompletionProposal[] computeCompletionProposals(ITextViewer viewer, int
 					continue;
 				}
 				if (element.regionMatches(true, 0, value, 0, value.length())) {
-					TypeCompletionProposal proposal = new TypeCompletionProposal(element, null, element, lineStart,
+					Image img = getImage(element);
+					TypeCompletionProposal proposal = new TypeCompletionProposal(element, img, element, lineStart,
 							value.length());
 					completions.add(proposal);
-				}
 			}
-			return completions.toArray(ICompletionProposal[]::new);
-		} catch (BadLocationException e) {
+		}
+		return completions.toArray(ICompletionProposal[]::new);
+	} catch (BadLocationException e) {
+	}
+	return null;
+}
+
+public Image getImage(String element) {
+	ImageDescriptor desc = null;
+	switch (element)
+		{
+		case "src.includes", "src.excludes", "src.additionalRoots", "permissions", "root.", "root", ".permissions.", ".link", "folder.", ".folder.", "link", "source.", "sourceFileExtensions", "bin.includes", "bin.excludes", "javacCustomEncodings.", "javacDefaultEncoding." -> desc = PDEPluginImages.DESC_CATEGORY_OBJ;  //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$ //$NON-NLS-4$ //$NON-NLS-5$ //$NON-NLS-6$ //$NON-NLS-7$ //$NON-NLS-8$//$NON-NLS-9$ //$NON-NLS-10$ //$NON-NLS-11$ //$NON-NLS-12$ //$NON-NLS-13$ //$NON-NLS-14$ //$NON-NLS-15$ //$NON-NLS-16$ //$NON-NLS-17$
+		case ".jar", "jars.compile.order", "jars.extra.classpath" -> desc = PDEPluginImages.DESC_JAR_LIB_OBJ; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+		case "javacProjectSettings" -> desc = PDEPluginImages.DESC_SETTINGS_OBJ; //$NON-NLS-1$
+		case "javacErrors." -> desc = PDEPluginImages.DESC_ERROR_ST_OBJ; //$NON-NLS-1$
+		case "significantVersionDigits", "generatedVersionLength" -> desc = PDEPluginImages.DESC_INFO_ST_OBJ; //$NON-NLS-1$ //$NON-NLS-2$
+		case "javacWarnings." -> desc = PDEPluginImages.DESC_ALERT_OBJ; //$NON-NLS-1$
+		case "jre.compilation.profile" -> desc = PDEPluginImages.DESC_TARGET_ENVIRONMENT; //$NON-NLS-1$
+		case "manifest." -> desc = PDEPluginImages.DESC_FOLDER_OBJ; //$NON-NLS-1$
+		default -> desc = PDEPluginImages.DESC_DEFAULT_OBJ;
+
+			}
+		if (desc != null) {
+			return desc.createImage();
 		}
 		return null;
```

Comment:
You can use the switch itself as an expression instead of assigning the value of the `desc` variable in each case:
```suggestion
	public Image getImage(String element) {
		@SuppressWarnings("nls")
		ImageDescriptor desc = switch (element)
			{
			case "src.includes", "src.excludes", "src.additionalRoots", "permissions", "root.", "root", ".permissions.", ".link", "folder.", ".folder.", //
					"link", "source.", "sourceFileExtensions", "bin.includes", "bin.excludes", "javacCustomEncodings.", "javacDefaultEncoding." -> PDEPluginImages.DESC_CATEGORY_OBJ;
			case ".jar", "jars.compile.order", "jars.extra.classpath" -> PDEPluginImages.DESC_JAR_LIB_OBJ;
			case "javacProjectSettings" -> PDEPluginImages.DESC_SETTINGS_OBJ;
			case "javacErrors." -> PDEPluginImages.DESC_ERROR_ST_OBJ;
			case "significantVersionDigits", "generatedVersionLength" -> PDEPluginImages.DESC_INFO_ST_OBJ;
			case "javacWarnings." -> PDEPluginImages.DESC_ALERT_OBJ;
			case "jre.compilation.profile" -> PDEPluginImages.DESC_TARGET_ENVIRONMENT;
			case "manifest." -> PDEPluginImages.DESC_FOLDER_OBJ;
			default -> PDEPluginImages.DESC_DEFAULT_OBJ;
			};
		return desc.createImage();
	}
```
Furthermore in this case I would simply suppress the nls warnings instead of adding $NON-NLS-1$ comments up to level 17 and split the line manually to make it readable.

Reply:
@HannesWell 
I have modified the switch case and suppressed the nls warnings. 
After splitting the lines manually though, it reverts back to the original form upon formatting. 
Do let me know in case of any more edits.